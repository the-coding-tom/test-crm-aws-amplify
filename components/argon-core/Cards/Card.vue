<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
