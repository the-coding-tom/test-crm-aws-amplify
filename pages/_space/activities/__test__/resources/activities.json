{
  "data": [
    {
      "id": 215,
      "log_name": "default",
      "description": "Kidd Bubu booked Conference Rooms",
      "subject_id": 68,
      "subject_type": "App\\RoomBooking",
      "causer_id": 27,
      "causer_type": "App\\User",
      "properties": { "attributes": { "space_id": 1 } },
      "created_at": "2020-03-24T19:17:38.000000Z",
      "updated_at": "2020-03-24T19:17:38.000000Z"
    },
    {
      "id": 214,
      "log_name": "default",
      "description": "Kidd Bubu purchased a ticket for The Next Frontier",
      "subject_id": 9,
      "subject_type": "App\\Event",
      "causer_id": 27,
      "causer_type": "App\\User",
      "properties": { "attributes": { "space_id": 1 } },
      "created_at": "2020-03-23T16:54:03.000000Z",
      "updated_at": "2020-03-23T16:54:03.000000Z"
    }
  ],
  "meta": {
    "current_page": 1,
    "from": 1,
    "last_page": 14,
    "path": "https://staging.bdgic76uiy.us-east-1.elasticbeanstalk.com/api/v1/san-francisco/activities",
    "per_page": 15,
    "to": 15,
    "total": 209
  },
  "links": {
    "first": "https://staging.bdgic76uiy.us-east-1.elasticbeanstalk.com/api/v1/san-francisco/activities?page=1",
    "last": "https://staging.bdgic76uiy.us-east-1.elasticbeanstalk.com/api/v1/san-francisco/activities?page=14",
    "prev": null,
    "next": "https://staging.bdgic76uiy.us-east-1.elasticbeanstalk.com/api/v1/san-francisco/activities?page=2"
  }
}
