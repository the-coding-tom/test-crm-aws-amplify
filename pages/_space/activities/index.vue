<template>
  <!--Charts-->
  <div class="container-fluid mt--6">
    <div class="card-deck flex-column flex-xl-row">
      <card>
        <div
          slot="header"
          class="row align-items-center">
          <div class="col">
            <SectionTitle
              title="Activity Feed"
              subtitle="Usage Stream on Mobile Platform"/>
          </div>
        </div>
        <div class="m-n25 sh-activity">
          <table class="table table-hover table-striped">
            <tbody>
              <EmptyActivity v-if="data.length == 0" />
              <template v-else>
                <Activity
                  v-for="activity in data"
                  :key="activity.id"
                  :date="getTimestamp(activity)">
                  {{ activity.description }}
                </Activity>
              </template>
            </tbody>
          </table>
        </div>
        <div
          slot="footer"
          class="text-center">
          <b-button
            variant="link"
            class="mr-r-30"
            @click="prev"
          >&lsaquo; Previous</b-button>  <b-button
            variant="link"
            class="mr-l-30"
            @click="next"
          > Next &rsaquo;</b-button>
        </div>
      </card>
    </div>
  </div>

</template>
<script>
import Activity from '~/components/shack/Activity.vue'
import MainTitle from '~/components/shack/MainTitle.vue'
import SectionTitle from '~/components/shack/SectionTitle.vue'
import EmptyActivity from '~/components/shack/EmptyActivity.vue'
import FeedTable from '~/components/shack/FeedTable.vue'
import Booking from '~/components/shack/Booking.vue'
import mz from 'moment-timezone'
import { mapState } from 'vuex'
import axios from 'axios'

export default {
  layout: 'ShackDash',
  async asyncData({ $activity, error }) {
    return await $activity
      .getAllActivities()
      .then(({ data, meta, links }) => {
        return {
          data,
          links,
          meta
        }
      })
      .catch(e => {
        const message = e.response
          ? `${e.response.data.message} ~ ${JSON.stringify(
              e.response.data.errors
            )}`
          : e.message
        error({ statusCode: e.statusCode, message })
      })
  },
  components: {
    Activity,
    Booking,
    FeedTable,
    EmptyActivity,
    SectionTitle,
    MainTitle
  },
  data() {
    return {
      data: []
    }
  },
  computed: {
    ...mapState({
      space: state => state.space.currentSpace
    })
  },
  methods: {
    getTimestamp(activity) {
      return mz(activity.created_at)
        .tz(this.space.timezone)
        .fromNow()
    },
    next() {
      this.getActivity(this.links.next)
    },
    prev() {
      this.getActivity(this.links.prev)
    },
    getActivity(link) {
      this.$axios
        .$get(link)
        .then(res => {
          this.data = res.data
          this.meta = res.meta
          this.links = res.links
        })
        .catch(e => {
          const message = e.response
            ? `${e.response.data.message} ~ ${JSON.stringify(
                e.response.data.errors
              )}`
            : e.message
          this.$bvToast.toast(message, {
            variant: 'error'
          })
        })
    }
  }
}
</script>
