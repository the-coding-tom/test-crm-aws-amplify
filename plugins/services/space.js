export default function(ctx, inject) {
  const Space = {
    updateOneSpace: (spaceId, payload) => {
      return ctx.$axios.$patch(`/spaces/${spaceId}`, payload)
    }
  }
  ;(ctx.$space = Space), inject('space', Space)
}
