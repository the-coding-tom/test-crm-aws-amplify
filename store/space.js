export const state = () => ({
  currentSpace: null
})

export const mutations = {
  setSpace(state, spaceDetails) {
    state.currentSpace = spaceDetails
  }
}
