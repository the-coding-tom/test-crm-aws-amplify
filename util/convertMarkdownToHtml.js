export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
