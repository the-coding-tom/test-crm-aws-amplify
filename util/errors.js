export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
