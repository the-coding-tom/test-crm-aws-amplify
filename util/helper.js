const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
