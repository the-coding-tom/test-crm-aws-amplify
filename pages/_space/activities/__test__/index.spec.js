import Card from '~/components/argon-core/Cards/Card.vue'
import IndexPage from '../index.vue'
import BootstrapVue from 'bootstrap-vue'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'

import space from '@/pages/_space/__test__/resources/current-space.json'
import activities from './resources/activities.json'
import nextActivities from './resources/next-activities.json'
import prevActivities from './resources/prev-activities.json'
import emptyActivities from './resources/empty-activities.json'

import { createWrapper } from '@/test.util.js'

describe('Page: /:space/activities/ | Activities Index Page', () => {
  let NuxtStore, store, wrapper, $activity, $axios, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)
    createRouter()

    $activity = {
      getAllActivities: jest
        .fn()
        .mockResolvedValueOnce(cloneDeep(emptyActivities))
        .mockResolvedValue(cloneDeep(activities))
    }

    $axios = {
      $get: jest
        .fn()
        .mockResolvedValueOnce(cloneDeep(nextActivities))
        .mockResolvedValueOnce(cloneDeep(prevActivities))
    }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', cloneDeep(space))

    const asyncDataResponse = await IndexPage.asyncData({ $activity })

    router = new VueRouter({})

    wrapper = createWrapper(
      IndexPage,
      store,
      router,
      asyncDataResponse,
      {},
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(Card.name, Card)
        vueInstance.prototype.$activity = $activity
        vueInstance.prototype.$axios = $axios
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('shows all activites', async () => {
    const wrapArr = wrapper.findAll('td.activity')

    expect(wrapArr.length).toBe(2)
    expect(wrapArr.at(0).text()).toBe(activities.data[0].description)
    expect(wrapArr.at(1).text()).toBe(activities.data[1].description)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders next page', async () => {
    wrapper.find('button.btn.mr-l-30.btn-link').trigger('click')
    await flushPromises()

    const wrapArr = wrapper.findAll('td.activity')

    expect(wrapArr.length).toBe(1)
    expect(wrapArr.at(0).text()).toBe(nextActivities.data[0].description)
  })

  it('renders prev page', async () => {
    wrapper.find('button.btn.mr-r-30.btn-link').trigger('click')
    await flushPromises()

    const wrapArr = wrapper.findAll('td.activity')

    expect(wrapArr.length).toBe(1)
    expect(wrapArr.at(0).text()).toBe(prevActivities.data[0].description)
  })
})
