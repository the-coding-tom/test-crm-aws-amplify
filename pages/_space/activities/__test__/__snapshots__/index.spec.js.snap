// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/activities/ | Activities Index Page renders properly 1`] = `
<div class="container-fluid mt--6">
  <div class="card-deck flex-column flex-xl-row">
    <div class="card">
      <div class="card-header">
        <div class="row align-items-center">
          <div class="col">
            <h3 class="page-title"><span>Activity Feed</span> <b>•</b> Usage Stream on Mobile Platform</h3>
          </div>
        </div>
      </div>
      <div class="card-body">
        <div class="m-n25 sh-activity">
          <table class="table table-hover table-striped">
            <tbody>
              <tr>
                <td> No activity recorded on platform</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <!---->
      <div class="card-footer">
        <div class="text-center"><button type="button" class="btn mr-r-30 btn-link">‹ Previous</button> <button type="button" class="btn mr-l-30 btn-link"> Next ›</button></div>
      </div>
    </div>
  </div>
</div>
`;
